syncer/syncer_test.go

```diff
@@ -49,6 +50,7 @@ type testSyncerSuite struct {
 	syncer   *replication.BinlogSyncer
 	streamer *replication.BinlogStreamer
 	cfg      *config.SubTaskConfig
+	cpMock   sqlmock.Sqlmock
```

Comment:
we can extract checkpoint's test cases into another suite like `testCheckpointSuite` later.

Reply:
moved into separate testSuite